File: fleet-pay-source-code 1 (1)/fleet-pay-export/flutter/lib/models/system_log.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'user.dart';
// --- start import code ---
  
// --- end import code ---

part 'system_log.g.dart';
part 'system_log.freezed.dart';

@unfreezed
class SystemLog with _$SystemLog {
  // --- start init code ---
  
  // --- end init code ---

  factory SystemLog({
    required String id,
    required String userId,
    required String companyId,
    User? user,
    required String action,
    required String message,
    required DateTime createdAt,
  }) = _SystemLog;

  factory SystemLog.fromJson(Map<String, dynamic> json) => _$SystemLogFromJson(json);

  // --- start extension code ---
  
  // --- end extension code ---

}

File: 1 fleet-pay-export/api/src/models/system_log.ts
import { User } from './user';

export interface SystemLog {
  id: string,
  userId: string,
  companyId: string,
  user?: User | null,
  action: string,
  message: string,
  createdAt: Date,
  // --- start extra_fields code ---
  
  // --- end extra_fields code ---
}
